### PR TITLE
Added ability for ReflectUtil to instantiate an object using Companion.apply(String)

### PR DESCRIPTION
**Pull request by tfenne**
A fairly small change @nh13 once I figured it out with some help from Google.

### CODE REVIEW

File: commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala [hunk 1]
```diff
@@ -369,10 +369,17 @@ object ReflectionUtil {
...
+      Try {
+        import java.lang.reflect.Constructor
+        val ctor: Constructor[_] = clazz.getDeclaredConstructor(classOf[String])
+        ctor.setAccessible(true)
+        ctor.newInstance(value)
+      }.getOrElse {
+        val typ             = mirror.classSymbol(clazz).toType
+        val companionMirror = mirror.reflectModule(typ.typeSymbol.companion.asModule)
+        val companion       = companionMirror.instance
+        companion.getClass.getMethod("apply", classOf[String]).invoke(companion, value)
```
Comment (nh13) [inline commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala]: Just a comment or two, so we remember what this does when it is not fresh in our minds.

Comment (nh13) [reply commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala]: Also, should we add documentation to `constructFromString` and other public methods that there needs to be a string constructor or `apply(str: String)` method in the companion?

**Review by nh13 — APPROVED** [review]
Some small suggestions on comments and documentation, otherwise 👍

File: commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala (hunk 1 above)
Comment (tfenne) [reply commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala]: Comment added, and I beefed up the scaladoc on both this method and the public one at the top of the calltree.

**Pull request merged**